airflow/models/dagrun.py

```diff
@@ -431,6 +432,7 @@ def verify_integrity(self, session=None):
         # check for removed or restored tasks
         task_ids = []
         for ti in tis:
+            task_instance_mutation_hook(ti)
```

Comment:
Why do we need it here? this TIs are received from DB.

I think we should only mutate before storing it to DB

Reply:
For cases that we want to mutate task instances before each executions. For example, re-routing to different queue for retries.

Reply:
But you are already adding it before saving it to DB on L468, so is it needed here too is my question?

Reply:
L468 is reached when the task instance does not exist thus you create the task instance. However, during retries, the task instance already exists so it won't reach L468

Reply:
aah yes if you want to have a different queue for TI's that are up for retry this does makes sense